tests/test_packages/test_protocols/test_fipa.py

```diff
@@ -290,16 +290,203 @@ def test_fipa_decoding_unknown_performative():
+                                                                 FIPADialogue.EndState.DECLINED_CFP: 1}
+        assert self.dialogues.dialogues_as_seller is not None
+        msg = FIPAMessage(message_id=0,
+                          dialogue_reference=(str(0), ''),
+                          target=1,
+                          performative=FIPAMessage.Performative.ACCEPT)
+        msg.counterparty = "opponent"
+        result = self.dialogues.is_permitted_for_new_dialogue(msg)
+        assert result is False
+
+    def test_dialogues_is_belonging_to_other_initiated_dialogue_label(self):
+        """Test if the given dialogue belongs to an other initiated dialogue."""
+        result = self.dialogues.create_opponent_initiated(dialogue_opponent_addr="opponent",
+                                                          dialogue_reference=(str(0), ''), is_seller=True)
+        result._outgoing_messages.append(self.last_msg)
+        other_initiated_dialogue = DialogueLabel((str(0), ''), "opponent", "opponent")
+
+        self.dialogues.dialogues[other_initiated_dialogue] = cast(FIPADialogue, result)
+        msg = FIPAMessage(message_id=1,
+                          dialogue_reference=(str(0), ''),
+                          target=1,
+                          performative=FIPAMessage.Performative.ACCEPT_W_INFORM,
+                          info={"address": "dummy_address"})
+        msg.counterparty = "opponent"
+        response = self.dialogues.is_belonging_to_registered_dialogue(msg, agent_addr="opponent")
+        assert response, "We expect the response from the function to be true."
+
+    def test_dialogues_is_belonging_to_self_initiated_dialogue_label(self):
+        """Test if the given dialogue belongs to self initiated dialogue."""
+        result = self.dialogues.create_self_initiated(dialogue_starter_addr="starter", dialogue_opponent_addr="opponent",
+                                                      is_seller=False)
+        result._outgoing_messages.append(self.last_msg)
+        self_initiated_dialogue_label = DialogueLabel((str(0), ''), "opponent", "starter")
+        self.dialogues.dialogues[self_initiated_dialogue_label] = cast(FIPADialogue, result)
+        msg = FIPAMessage(message_id=1,
+                          dialogue_reference=(str(0), ''),
+                          target=1,
+                          performative=FIPAMessage.Performative.ACCEPT_W_INFORM,
+                          info={"address": "dummy_address"})
+        msg.counterparty = "opponent"
+        response = self.dialogues.is_belonging_to_registered_dialogue(msg, agent_addr="starter")
+        assert response, "We expect the response from the function to be true"
+
+    def test_dialogues_is_belonging_to_alternative_initiated_dialogue_label(self):
+        """Test if the given dialogue belongs to alternative initiated dialogue label."""
+        result = self.dialogues.create_self_initiated(dialogue_starter_addr="starter",
```

Comment:
all self initiated dialogues have an alternative dialogue label, so this test can be dropped.